mithril-common/src/messages/aggregator_features.rs

```diff
@@ -53,9 +58,40 @@ pub struct CardanoTransactionsProverCapabilities {
 
 #[cfg(test)]
 mod tests {
+    use crate::entities::BlockNumber;
+
     use super::*;
 
-    fn golden_message() -> AggregatorFeaturesMessage {
+    #[derive(Debug, Serialize, Deserialize, PartialEq)]
+    struct AggregatorFeaturesMessagePrevious {
+        pub open_api_version: String,
+        pub documentation_url: String,
+        pub capabilities: AggregatorCapabilitiesPrevious,
+    }
+
+    #[derive(Debug, Serialize, Deserialize, PartialEq)]
+    struct AggregatorCapabilitiesPrevious {
+        pub signed_entity_types: BTreeSet<SignedEntityTypeDiscriminants>,
+        #[serde(skip_serializing_if = "Option::is_none")]
+        pub cardano_transactions_prover: Option<CardanoTransactionsProverCapabilities>,
+    }
+
+    fn golden_message_v1() -> AggregatorFeaturesMessagePrevious {
```

Comment:
There is no need to test that the previous version (v1) is deserialized from a JSON v1.
It was already done when this version was the current one. 
What we really need is if a JSON v1 can be deserialized into the current structure version or if the new structure version can produce a JSON v1 (it can be the case with new optional field that will be filled later). 

| Aggregator  | Signer |  |
| -------------- | -------- | --- |
| V1 | V1 | already tested in previous code version |
| V1 | V2 | not supported |
| V2 | V1 | if json is send to aggregator => can we deserialize a V1,  if json is send to signer => can we deserialize a json V2 into previous struct | 
| V2 | V2 | - |